### PR TITLE
Cache local timezone to improve performance

### PR DESCRIPTION
**Pull request by nicksanders**
#68 

**Before**
5000 x Date.local takes 23073.4ms

**After**
5000 x Date.local takes 185.3ms

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -12,7 +12,7 @@ defmodule Timex.Mixfile do
   end
 
   def application do
-    [included_applications: [:tzdata]]
+    [included_applications: [:tzdata], timezone: nil]
```
Comment (bitwalker) [inline mix.exs]: Is this necessary to add? `get_env` and `put_env` should work fine without listing it here.

File: lib/timezone/timezone_local.ex [hunk 2]
```diff
@@ -44,13 +44,18 @@ defmodule Timex.Timezone.Local do
 
   def lookup(), do: Date.now |> lookup
   def lookup(%DateTime{} = date) do
-    case :os.type() do
-      {:unix, :darwin} -> localtz(:osx, date)
-      {:unix, _}       -> localtz(:unix, date)
-      {:nt}            -> localtz(:win, date)
-      {:win32, :nt}    -> localtz(:win, date)
-      _                -> raise "Unsupported operating system!"
+    tz = Application.get_env(:timex, :timezone)
```
Comment (bitwalker) [inline lib/timezone/timezone_local.ex]: I think we should change it from `:timezone` to `:local_timezone` just to be specific as to what is being set here.

**Comment by bitwalker** [comment]
Looks good! Just have a few comments to address and I can merge this :).

I'm sure I was caching this before, which is why I suspect it slowed down drastically after it was removed, I just can't recall the specific reasoning as to why.

**Comment by nicksanders** [comment]
Looks like version 0.13.5 was using 

Process.get(:local_timezone) and Process.put(:local_timezone, tz)

is that better than Application.get_env/env_put?

**Comment by bitwalker** [comment]
Application.get_env is a better place to put this honestly, the process dictionary isn't the ideal place for this after thinking about it some more. Thanks for making the changes so quickly!

**Pull request merged**